Yoast/wordpress-seo
# 14845 Remove media purge related stuff

## Context


* This is a CS tasks from the Cleanup sheet. This PR aims to remove the "media purge" class which is still instantiated as an integration but seems no longer necessary.

A few things that need to be checked with architects and product:
- The whole "media purge" class was introduced to remediate to a bug in Yoast SEO 7.0 (March 2018). The related notification is hidden by default since #10107. Therefore the whole thing seems no longer used and I guess we don't support upgrades from version 7.0. For more details, please see the related issue https://github.com/Yoast/wordpress-seo/issues/14845
- Is the [Search Index Purge](https://wordpress.org/plugins/yoast-seo-search-index-purge/) plugin still relevant? We should consider whether it's okay to retire it.
- This PR removes the `WPSEO_Admin_Media_Purge_Notification`. I guess we don't need to deprecate it, as this class wasn't meant to be used by other plugins or extended in any way. Worth having a confirmation about this point.


## Summary


This PR can be summarized in the following changelog entry:

* Removes the no longer used Admin Media Purge Notification class.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* build the plugin
* go to SEO > Dashboard and see there are no errors / warnings / notices
* navigate through the admin pages and see there are no errors / warnings / notices


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* go to SEO > Dashboard and see there are no errors / warnings / notices
* navigate through the admin pages and see there are no errors / warnings / notices

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes #14845
